sdk/compiler/damlc/tests/daml-test-files/HexString.daml

```diff
@@ -0,0 +1,142 @@
+-- Copyright (c) 2020, Digital Asset (Switzerland) GmbH and/or its affiliates.
+-- All rights reserved.
+
+-- @SUPPORTS-LF-FEATURE DAML_CCTP
+
+module HexString where
+
+import DA.Assert ((===))
+import DA.Optional (fromSome)
+import DA.Crypto.Text
+import Daml.Script
+
+n3: Numeric 3
+n3 = 0.123
+
+n7: Numeric 7
+n7 = 0.12345_67
+
+n10: Numeric 10
+n10 = 0.12345_67890
+
+main =
+  script do
+    -- `HasToHex Int` and `HasFromHex Int`
+    toHex 0 === "00"
+    toHex 1 === "01"
+    toHex 15 === "0f"
+    toHex 16 === "10"
+    toHex 255 === "ff"
+    toHex 256 === "0100"
+    toHex 257 === "0101"
+    toHex 3735928559 === "deadbeef"
+    fromHex "" === Some 0
+    fromHex "00" === Some 0
+    fromHex "01" === Some 1
+    fromHex "0a" === Some 10
+    fromHex "0F" === Some 15
+    fromHex "10" === Some 16
+    fromHex "Ff" === Some 255
+    fromHex "0100" === Some 256
+    fromHex "0101" === Some 257
+    fromHex "DeadBeef" === Some 3735928559
```

Comment:
we need test for all the exceptions cases.

Reply:
Currently planning to manage this on https://github.com/digital-asset/daml/issues/20947